docs/rfc/002_standard_memo_format.md

```diff
@@ -0,0 +1,131 @@
+<h1 style="text-align: center;">A proposed standard memo format for non-EVM chains</h1>
+<br>
+
+## The goal
+
+This doc is aimed to introduce a standard format of ZetaChain inbound `memo` across non-EVM chains. The `memo` in an inbound transaction is to carry both instructions and data, by which the cross-chain transaction to be processed as expected in the `zetacore`. A standard memo format helps to ensure consistent developer experience across EVM and non-EVM chains.
+<br><br>
+
+
+## The EVM gateway inbound events
+
+We want to mimic below `Deposit` and `Called` events in the EVM `gateway` contract for non-EVM chains. Typically, we at least want the following 3 pieces at the time of writing.
+
+- The `receiver` address in ZEVM.
+- The `payload` of the call.
+- The `RevertOptions`.
+
+```
+event Deposited(
+    address indexed sender,
+    address indexed receiver,
+    uint256 amount,
+    address asset,
+    bytes payload,
+    RevertOptions revertOptions
+);
+
+event Called(
+	   address indexed sender,
+	   address indexed receiver,
+	   bytes payload,
+	   RevertOptions revertOptions
+);
+```
+<br><br>
+
+
+## 1. The Memo Diagram
+
+![](images/002_layout.png)<br><br>
+
+
+## 2. The memo structure
+
+The raw-byte representation of a standard non-EVM memo contains `9` sections (2 mandatory + 7 optional). 
+
+`Section 0` is the `header`, which controls how the memo should be interpreted.
+
+`Section 1~8` are data sections that carry actual data attached to the memo.
+
+
+|                 | Section 0  | Section 1   | Section 2   | Section 3      | Section 4    | Section 5    | Section 6     | Section 7   | Section 8   |
+|-----------------|------------|-------------|-------------|----------------|--------------|--------------|---------------|-------------|-------------|
+| **Name**        | Header     | Receiver    | Payload     | RevertAddress  | CallOnRevert | AbortAddress | RevertMessage | reserved    | reserved    |
+| **Size (byte)** | 3          | 20          | variable    | variable       | 1            | 20           | variable      | --          | --          |
+| **Type**        | [3]byte    | ZEVM address| bytes       | non-EVM address| bool         | ZEVM address | bytes         | --          | --          |
```

Comment:
Just for my understanding, why can't we use the receiver as the AbortAddress?

Reply:
A user would not want to have funds deposited in a smart contract if the smart contract call fail

Reply:
I think it's to give dev the a choice to send the aborted funds to a different account (may be same) if there is a need to handle aborted funds in a different way. I think @lumtis or @skosito had the original idea?